fiftyone/core/odm/database.py

```
@@ -847,7 +851,13 @@ def bulk_write(ops, coll, ordered=False, progress=False):
         with batcher:
             for batch in batcher:
                 batch = list(batch)
-                coll.bulk_write(batch, ordered=ordered)
+                res = coll.bulk_write(batch, ordered=ordered)
+                if res.bulk_api_result.get("nBytes") and hasattr(
+                    batcher, "set_encoding_ratio"
+                ):
+                    batcher.set_encoding_ratio(
```

super edge case that idk if possible, but can nBytes == 0 and possibly throw a division by zero error?

---

I added an if statement for this actually in the batcher

```
    def set_encoding_ratio(self, encoded_batch_size):
        if self._last_batch_content_size and encoded_batch_size > 0:
            self._encoding_ratio = (
                self._last_batch_content_size / encoded_batch_size
            )
```

this was mostly necessary because we are setting to -1 in the case that instert_many throws and we swallow the exception so we know that it might have gone through anyways and leave the encoding size as is.